ckan_datapackage_tools/converter.py

```diff
@@ -177,82 +174,169 @@ def _parse_notes(dataset_dict):
+    if licenses:
+        result['licenses'] = licenses
+    if contributors:
+        result['contributors'] = contributors
+    return result
 
-    if source:
-        result['sources'] = [source]
 
+def _parse_profile(dataset_dict):
+    result = {}
+    profile = dataset_dict.pop('profile', '')
+    if profile:
+        result['profile'] = profile
     return result
 
 
-def _parse_maintainer(dataset_dict):
-    result = {}
-    author = {}
+def _parse_organization_as_primary_source(dataset_dict):
+    all_sources = []
+    source = {}
+    organization = dataset_dict.get('organization')
+    if organization.get('is_organization', False):
+        source['title'] = organization.get('name', organization.get('title'))
+    if source.get('title'):
+        if dataset_dict.get('url'):
+            source['path'] = dataset_dict['url']
+        all_sources.append(source)
+    return all_sources
 
-    if dataset_dict.get('maintainer'):
-        author['name'] = dataset_dict['maintainer']
-    if dataset_dict.get('maintainer_email'):
-        author['email'] = dataset_dict['maintainer_email']
 
-    if author:
-        result['author'] = author
+def _parse_extra_sources(extras_dict, sources):
+    for source in extras_dict.pop('sources', []):
+        parsed = _parse_extra_source(source)
+        if parsed and parsed.get('title'):
+            sources.append(parsed)
 
-    return result
 
+def _parse_extra_source(extra_source):
+    source = {}
+    if extra_source.get('name'):
+        extra_source['title'] = extra_source.pop('name')
+    if extra_source.get('url'):
+        extra_source['path'] = extra_source.pop('url')
+    for prop in ['title', 'path', 'email']:
+        if extra_source.get(prop):
+            source[prop] = extra_source[prop]
+    return source
+
+
+def _parse_primary_license(dataset_dict):
+    licenses = []
+    license = _parse_license(dataset_dict)
+    if license:
+        licenses.append(license)
+    return licenses
 
-def _parse_tags(dataset_dict):
-    result = {}
 
-    keywords = [tag['name'] for tag in dataset_dict.get('tags', [])]
+def _parse_extra_licenses(extras_dict, licenses):
+    for license in extras_dict.pop('licenses', []):
+        parsed = _parse_extra_license(license)
+        if parsed:
+            if parsed.get('name') or parsed.get('path'):
+                licenses.append(parsed)
 
-    if keywords:
-        result['keywords'] = keywords
 
-    return result
+def _parse_license(dataset_dict):
+    license = {}
+    if dataset_dict.get('license_id'):
+        license['name'] = dataset_dict['license_id']
+    if dataset_dict.get('license_url'):
+        license['path'] = dataset_dict['license_url']
+    if dataset_dict.get('license_title'):
+        if license.get('name') or license.get('path'):
+            license['title'] = dataset_dict['license_title']
+    return license
 
 
-def _parse_extras(dataset_dict):
-    result = {}
+def _parse_extra_license(extra_license):
+    license = {}
```

Comment:
Similarly to my comment regarding `sources` above, I'd expect whatever is stored in the `licenses` extra to be the same that the DP so it should not be necessary to parse or map it.

Reply:
OK, so again just to be clear (for me ;) ), the ingest of a datapackage licenses should actually create the same in extras, but in addition, also push the first license to dataset's license (so that  this first license is captured in both dataset license and extras?

Reply:
Yes, that's correct. See my other comment above